src/GUI/EFCorePowerTools/EFCorePowerToolsPackage.cs

```diff
@@ -245,15 +245,18 @@ private IServiceProvider CreateServiceProvider()
                     .AddTransient<IPickServerDatabaseDialog, PickServerDatabaseDialog>()
                     .AddTransient<IPickTablesDialog, PickTablesDialog>()
                     .AddTransient<IModelingOptionsDialog, EfCoreModelDialog>()
-                    .AddTransient<IMigrationOptionsDialog, EfCoreMigrationsDialog>();
+                    .AddTransient<IMigrationOptionsDialog, EfCoreMigrationsDialog>()
+                    .AddTransient<IPickSchemasDialog, PickSchemasDialog>()
+                    .AddSingleton<Func<IPickSchemasDialog>>(sp => sp.GetService<IPickSchemasDialog>);
```

Comment:
Why is this required for the new dialog?

Reply:
To avoid ServiceLocator (anti)pattern.
If you inject directly `IPickSchemasDialog` into `IPickServerDatabaseViewModel` it works only once and you got an error if you close and re-open Schemas dialog (without closing PickServerDatabaseDialog). This happens because in `IPickServerDatabaseViewModel` only a single istance of `IPickSchemasDialog` is created; when this instance is closed invoking `CloseRequested` you cannot use it anymore. To create a new istance every time dialog is requested (like you already do with other dialogs) a factory function is the only solution.